### DISCUSSION TITLE
Returning custom messages for a model via the request file isn't working, any tips?

### DISCUSSION THREAD

**Discussion by adam-jones-net** [area17/twill#975] [Q&A]
I have a model that uses a repeater block, some media.   The only requirements is that some fields are 'required'.    The site has no multi-lingual support, so that keeps it all fairly simple. 

The _required_ setting works for all the fields I need it on, BUT no custom messages appear in Twill when I submit a form with empty _required_ fields that are within the repeater element I have.  Instead I get a generic:

"Your submission could not be validated, please fix and retry"

Is anything obviously wrong with the below ?

```
public function rulesForUpdate(){
        return [
            'address_summary' => 'required',
            'postcode' => 'required',
            'intro_copy' => 'required',
            'price_from' => 'numeric|min:1|max:50',
            'things_to_do_intro' => 'max:750',
            'things_to_do_shopping' => 'max:750',
            'things_to_do_fooddrink' => 'max:750',
            'things_to_do_nightlife' => 'max:750',
            'seo_title' => 'max:150',
            'seo_desc' => 'max:250',
            'description' => 'max:5000',
            'medias.image_featured' => 'required',
            'medias.image_gallery' => 'required',
            'browsers.locations' => 'required',
            'repeaters.rate.*.sleeps' => 'required',
            'repeaters.rate.*.label' => 'required',
            'repeaters.rate.*.price_from' => 'required'
        ];
    }

    public function messages() {
        return [
            'intro_copy.required' => "Intro copy is required",
            'address_summary.required' => "The address summary is required",
            'postcode.required' => "The postcode is required",
            'browsers.locations.required' => "A location is required",
            'medias.image_featured.required' => 'The featured image is required',
            'medias.image_gallery.required' => 'At least one image in the image gallery is required',
            'rate.sleeps.required' => 'A room size is required',
            'rate.label.required' => 'required',
            'rate.price_from.required' => 'required'
        ];
    }
```

**Comment by pboivin** [comment]
Hey @arkid,

I don't see anything obviously wrong with your code snippet. I'm not 100% sure about the `repeaters.x.*.y` syntax but maybe someone else can comment on this.

In any case, even without customizing the messages, you should be getting a default error message for required fields :

![required-default](https://user-images.githubusercontent.com/7805679/122229610-e5b95500-ce86-11eb-8ce2-de216104a8b3.png)

Can you replicate this behavior on a simple module with one required field?

**Reply by adam-jones-net** [reply]
The problem appears to be just within the repeater block (I've updated question to reflect this).   Yes other fields seem to work ok with the red text below them.   However repeater blocks show no red text below themselves.

**Reply by pboivin** [reply]
@arkid I did some quick tests on my site and I think that it's indeed a bug.

**From the module's Request**

```
public function rulesForUpdate()
{
    return [
        'repeaters.postmedia.*.description' => 'required',
    ];
}

public function messages()
{
    return [
        'repeaters.postmedia.*.description.required' => 'MY CUSTOM ERROR MESSAGE',
    ];
}
```

**From the response tab in dev tools**

![2021-06-16_14-10](https://user-images.githubusercontent.com/7805679/122271203-2deb6e00-cead-11eb-87be-887b053d5371.png)

The custom error message is processed and returned to the frontend but it doesn't seem to reach the field inside the repeater.

The `repeaters.x.*.y` syntax is correct by the way, but you already knew that :)

**Reply by adam-jones-net** [reply]
Ok thanks for clarifying.   It was actually mentioned by myself (sorry I only have short term memory these days) as well as others in this [github issue](https://github.com/area17/twill/issues/697).

**Reply by pboivin** [reply]
Ah, that's good! Feel free to open a fresh issue to surface this bug more precisely.
